### PR TITLE
fix(learn): Added solution for invert binary tree data structures challenge

### LINKED ISSUE #35914: Incomplete Challenge: Invert a Binary Search Tree
**Describe your problem and how to reproduce it:**
In both the curriculum and guides directories, when you look in the invert-a-binary-tree.md and  index.md respectively, there is no solution and when I go to the challenge as a regular user of fcc, there is no invert method as the challenge description indicates to complete.

**Add a Link to the page with the problem:**
-https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/08-coding-interview-prep/data-structures/invert-a-binary-tree.english.md
-https://github.com/freeCodeCamp/freeCodeCamp/blob/master/guide/english/certifications/coding-interview-prep/data-structures/invert-a-binary-tree/index.md

**Tell us about your browser and operating system:**
* Browser Name: Google Chrome
* Browser Version: 73.0.3683.103
* Operating System: Ubuntu 16.04 LTS

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
Screenshot from 2019-04-25 12-44-32

Hi.  I am wondering if I can work on resolving this challenge e.g. solution possibly, make an invert method etc.  Thank you.

### PR DESCRIPTION
**Pull request by StanimalTheMan**
[ x ] I have read freeCodeCamp's contribution guidelines.
[ x ] My pull request has a descriptive title (not a vague title like Update index.md)
[ x ] My pull request targets the master branch of freeCodeCamp.
[ x ] None of my changes are plagiarized from another source without proper attribution.
[ x ] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
[ x ] My changes do not use shortened URLs or affiliate links.
Closes #35914

Maybe it's just me but I do not see a certifications subdirectory in the mock-guide directory anymore so I could not test on the local copy of fcc as I was told to here:

https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/how-to-open-a-pull-request.md
#35511

### CODE REVIEW

**Comment by sil3nthill** [comment]
hello , i think its better to test  run tests locally before ! 
with `npm run test `or `npm run test:curriculum` in your case .

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@StanimalTheMan In addition to what @camperbot pointed out, you need to make sure all solutions include all original challenge seed and any changes to solve the challenge.  Both of the solutions you provide in the files you changed are only partial solutions.

**Comment by Dharmin2108** [comment]
j

**Comment by StanimalTheMan** [comment]
Alright.  Thank you.

**Comment by StanimalTheMan** [comment]
@RandellDawson When/if you can, please let me know if there is still something wrong.  Thank you.

**Comment by RandellDawson** [comment]
@StanimalTheMan Would you mind using 2 space indentation per our style guides?  Other than that, I believe it looks good.

**Comment by StanimalTheMan** [comment]
@RandellDawson Alright, I think I followed what you told me to do.  Thank you.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
We're moving away from using emojis in the guide (see [here](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/style-guide-for-guide-articles.md#images)), but other than that everything looks good.

**Comment by StanimalTheMan** [comment]
@ojeytonwilliams thank you for the heads up. @RandellDawson Sorry for the late reply.  I see that you self-assigned the task; please let me know if you want me to still work on this.  Thank you.

**Comment by RandellDawson** [comment]
You still need to make the change @ojeytonwilliams recommended.

**Comment by ojeytonwilliams** [comment]
@StanimalTheMan yeah, I could have put it a bit better, but I did actually want you take the emojis out.

File: guide/english/certifications/coding-interview-prep/data-structures/invert-a-binary-tree/index.md [hunk 1]
```diff
@@ -2,9 +2,62 @@
 title: Invert a Binary Tree
 ---
 ## Invert a Binary Tree
+Remember to use <a>**`Read-Search-Ask`**</a> if you get stuck. Try to pair program
```
Comment (RandellDawson) [inline guide/english/certifications/coding-interview-prep/data-structures/invert-a-binary-tree/index.md]:
```suggestion
```

File: guide/english/certifications/coding-interview-prep/data-structures/invert-a-binary-tree/index.md [hunk 2]
```diff
@@ -2,9 +2,62 @@
 title: Invert a Binary Tree
 ---
 ## Invert a Binary Tree
+Remember to use <a>**`Read-Search-Ask`**</a> if you get stuck. Try to pair program 
 
-This is a stub. <a href='https://github.com/freecodecamp/guides/tree/master/src/pages/certifications/coding-interview-prep/data-structures/invert-a-binary-tree/index.md' target='_blank' rel='nofollow'>Help our community expand it</a>.
+### Problem Explanation:
```
Comment (RandellDawson) [inline guide/english/certifications/coding-interview-prep/data-structures/invert-a-binary-tree/index.md]: ``suggestion
```

File: guide/english/certifications/coding-interview-prep/data-structures/invert-a-binary-tree/index.md [hunk 3]
```diff
@@ -2,9 +2,62 @@
...
 
-<a href='https://github.com/freecodecamp/guides/blob/master/README.md' target='_blank' rel='nofollow'>This quick style guide will help ensure your pull request gets accepted</a>.
+To solve this problem, you should use a function to invert a binary tree.  A temp variable can possibly be used and a stack can work as well but doing so may require more memory.
```
Comment (RandellDawson) [inline guide/english/certifications/coding-interview-prep/data-structures/invert-a-binary-tree/index.md]:
```suggestion
```

File: guide/english/certifications/coding-interview-prep/data-structures/invert-a-binary-tree/index.md [hunk 4]
```diff
@@ -2,9 +2,62 @@
...
 
-<!-- The article goes here, in GitHub-flavored Markdown. Feel free to add YouTube videos, images, and CodePen/JSBin embeds  -->
+
```
Comment (RandellDawson) [inline guide/english/certifications/coding-interview-prep/data-structures/invert-a-binary-tree/index.md]:
```suggestion
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@StanimalTheMan My assignment just means I was going to be reviewing it. 

See my suggested changes as we have been removing the Problem Descriptions from these articles as it is just duplicated from the challenge itself.

**Review by ojeytonwilliams — APPROVED** [review]

**Review by RandellDawson — APPROVED** [review]

**Review by Manish-Giri — APPROVED** [review]
@StanimalTheMan LGTM. Thank You for your contribution.

I'll make a PR in a bit to fix some formatting issues in the guide.

**Pull request merged**